docs/source/tutorial/Introduction.lhs

```diff
@@ -12,63 +12,146 @@ module Introduction where
 
 import Protolude
 
+import System.Random
+
 import GraphQL
 import GraphQL.API (Object, Field, Argument, (:>), Union)
 import GraphQL.Resolver (Handler, (:<>)(..), unionValue)
 ```
 
-The core idea for this library is that we define a composite type that
-specifies the whole API, and then implement a matching handler.
+## A simple GraphQL service
 
-The main GraphQL entities we care about are Objects and Fields. Each
-Field can have arguments.
+A [GraphQL](http://graphql.org/) service is made up of two things:
 
-``` haskell
-type HelloWorld = Object "HelloWorld" '[]
-  '[ Argument "greeting" Text :> Field "me" Text
-   ]
-```
+ 1. A schema that defines the service
+ 2. Some code that implements the service's behavior
 
-The example above is equivalent to the following GraphQL type:
+We're going to build a very simple service that says hello to
+people. Our GraphQL schema for this looks like:
```

Comment:
maybe add a note about not confusing graphql types and haskell types? I got a bit confused with the string even though I wrote part of this :)